src/versionbits.cpp

```diff
@@ -74,16 +78,14 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
                 }
                 if (count >= nThreshold) {
                     stateNext = ThresholdState::LOCKED_IN;
-                } else if (pindexPrev->GetMedianTimePast() >= nTimeTimeout) {
-                    stateNext = ThresholdState::FAILED;
+                } else if((fHeightBased && (pindexPrev->nHeight + 1) >= nHeightTimeout) ||
+                      (!fHeightBased && pindexPrev->GetMedianTimePast() >= nTimeTimeout)) {
+                    stateNext = (fHeightBased == true) ? ThresholdState::LOCKED_IN : ThresholdState::FAILED;
                 }
                 break;
             }
             case ThresholdState::LOCKED_IN: {
-                // Progresses into ACTIVE provided activation height will have been reached.
-                if (pindexPrev->nHeight + 1 >= min_activation_height) {
-                    stateNext = ThresholdState::ACTIVE;
-                }
+                stateNext = ThresholdState::ACTIVE;
```

Comment:
why was this check removed for non height based?

Reply:
It was temporary code added for bitcoin's taproot activation that we weren't using. Removing it made the logic simpler, easier to read, and easier to test.